opencensus/stats/exporters/stackdriver_exporter.py

```diff
@@ -0,0 +1,385 @@
+DEFAULT_DISPLAY_NAME_PREFIX = "OpenCensus"
+ERROR_BLANK_PROJECT_ID = "expecting a non-blank ProjectID"
+CONS_NAME = "name"
+CONS_TIME_SERIES = "timeseries"
+EPOCH_DATETIME = datetime(1970, 1, 1)
+EPOCH_PATTERN = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+
+class Options(object):
+    """ Options contains options for configuring the exporter.
+    """
+    def __init__(self,
+                 project_id="",
+                 resource="",
+                 metric_prefix="",
+                 default_monitoring_labels=None):
+        self._project_id = project_id
+        self._resource = resource
+        self._metric_prefix = metric_prefix
+        self._default_monitoring_labels = default_monitoring_labels
+
+    @property
+    def project_id(self):
+        """ project_id is the identifier of the Stackdriver
+        project the user is uploading the stats data to.
+        If not set, this will default to
+        your "Application Default Credentials".
+        """
+        return self._project_id
+
+    @property
+    def resource(self):
+        """ Resource is an optional field that represents the Stackdriver
+        MonitoredResource type, a resource that can be used for monitoring.
+        If no custom ResourceDescriptor is set, a default MonitoredResource
+        with type global and no resource labels will be used.
+        Optional.
+        """
+        return self._resource
+
+    @property
+    def metric_prefix(self):
+        """ metric_prefix overrides the
+        OpenCensus prefix of a stackdriver metric.
+        Optional.
+        """
+        return self._metric_prefix
+
+    @property
+    def default_monitoring_labels(self):
+        """ default_monitoring_labels are labels added to
+        every metric created by this
+        exporter in Stackdriver Monitoring.
+
+        If unset, this defaults to a single label
+        with key "opencensus_task" and value "py-<pid>@<hostname>".
+        This default ensures that the set of labels together with
+        the default Resource (global) are unique to this
+        process, as required by Stackdriver Monitoring.
+
+        If you set default_monitoring_labels,
+        make sure that the Resource field
+        together with these labels is unique to the
+        current process. This is to ensure that
+        there is only a single writer to
+        each TimeSeries in Stackdriver.
+
+        Set this to Labels to avoid getting the
+        default "opencensus_task" label.
+        You should only do this if you know that
+        the Resource you set uniquely identifies this Python process.
+        """
+        return self._default_monitoring_labels
+
+
+class StackdriverStatsExporter(base.StatsExporter):
+    """ StackdriverStatsExporter exports stats
+    to the Stackdriver Monitoring."""
+    def __init__(self,
+                 options=Options(),
+                 client=None,
+                 default_labels={},
+                 transport=sync.SyncTransport):
```

Comment:
Could we please make the default we want is an AsyncTransport? Currently we have SyncTransport as the default
and it has caused a bit of unexpectedness where people perceive the libraries as being slow yet it is just the SyncTransport.

Reply:
Thanks for your reply. Done !